[12.0] mig account_financial_report

eLBati opened the pull request:
Rebuilt branch from 11.0 and applied commits from https://github.com/OCA/account-financial-reporting/pull/509

TODO
 - [x] https://github.com/OCA/account-financial-reporting/pull/509#pullrequestreview-206896535
 - [x] https://github.com/OCA/account-financial-reporting/pull/509#issuecomment-495672367
 - [x] https://github.com/OCA/account-financial-reporting/pull/555
 - [x] https://github.com/OCA/account-financial-reporting/pull/509#issuecomment-491489410
 - [x] https://github.com/OCA/account-financial-reporting/pull/559#issuecomment-497226416 (Fekete)
 - [x] Translations (https://github.com/OCA/account-financial-reporting/pull/559#issuecomment-497230293)

Commit:
[FIX] account_financial_report_qweb: date condition in open items report.
This commit includes move lines on the report date by default. These lines shouldn't be considered in the futur.

JordiBForgeFlow committed:
[MIG] account_financial_report_qweb: Migration to 11.0

* dynamize general ledger

oca-transbot committed:
OCA Transbot updated translations from Transifex

rgarnau committed:
[ADD] Responsive reports

Commit:
[10.0] AFR Aged partner balance : columns and datas consistency. (#383)

oca-transbot committed:
OCA Transbot updated translations from Transifex

feketemihai committed:
Update links in report, add account group file, update trial balance with hierarchy.

Update indentation, remove empty lines from header.

Update test.

Update pylint.

Remove company_id on computing accounts, since account.group is not a company based model, filtering accounts is done on trial balance report.

Update account variables.

Improve condition in padding on accounts.

Add option to print hierarchy based on defined accounts/computed accounts.

Add VAT report, hierarchy from tax tags ans taxes.

Fix pylint, xlsx report generation header.

Update code to select code_prefix or name.

Update code to select code_prefix or name.

Update code to select code_prefix or name.

Fix domain in base amounts in vat report.

Change trial balance code_prefix or name.

Update trail balance, add tests for vat report.

Update pylint, amounts as monetary, many2one option on generation excels.

Update pulint.

Add VAT Report in readme.

Add VAT Report in readme.

Update array_agg.

Update array_agg.

Update array_agg.

Add option in VAT Report to be printed on Tax Tags - Tax Groups.

Add widget to hierarchy_on on trial balance.

MiquelRForgeFlow committed:
[IMP] account_financial_report: adapt the wizards to multicompany

JordiBForgeFlow committed:
fixes the reporting of unaffected earnings account

HviorForgeFlow committed:
[11.0][IMP] account_financial_report - foreign_currency

osoul-dev committed:
Added translation using Weblate (Arabic)

Translated using Weblate (Arabic)

Currently translated at 95.7% (177 of 185 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ar/

BT-astauder committed:
Increasing version because of #PR403

Some changes in the databases been made in #403, but no migration is needed.

pedrobaeza committed:
[FIX] account_financial_report: Fix tests + sql for integration tests

When this module is installed along with other chart account different from generic one,
the number of expected accounts and the computation change (for example, in Spain,
the unaffected earnings account is 129000, choking with group with code prefix 1).

This commit makes the tests resistent to these changes.

mpanarin committed:
[ADD] 11.0 account_financial_report: add filter by journals

To general ledger

HviorForgeFlow committed:
move font-family from generic tags

ons-eco committed:
[11][FIX] Replace partner_ids by journal_ids

p-tombez committed:
[11.0][IMP] Improve transient models cleaning

When there are a lot of account.move.line (several millions) and print any of
the Qweb reports, that will generate also a lot of transient objects.
As these objects are created with an "insert" query, the cleaning normally
triggered by the count of the records in transient tables is not done, so only
the cleaning based on the age of the records is processed (by default, records
older than 1 hours are deleted), but the cron task is only ran one time per
day. For large setups this can lead to memory errors at that point. This change
prevents the memory error by executing the transient record cleanup for the
report models in this module in SQL.

rschnapka committed:
Translated using Weblate (German)

Currently translated at 73.9% (215 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/de/

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 50.2% (146 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

astirpe committed:
[FIX] _transient_clean_rows_older_than() WHERE clause

yaseentai committed:
Translated using Weblate (Arabic)

Currently translated at 96.2% (178 of 185 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ar/

jimmybanegas committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (291 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

Commit:
Translated using Weblate (Dutch)

Currently translated at 83.5% (243 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/nl/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (291 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

alessandrocamilli committed:
Translated using Weblate (Italian)

Currently translated at 23.0% (67 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/it/

JordiBForgeFlow committed:
[account_financial_report] adds the following features to Trial Balance:

- Adds 'Period balance' column
- Renames the option 'Hide accounts at 0'. Means no initial, no debit, credit
  or ending balance for the period.
- Fixes logic to remove lines with 0 activity for the period.
- improve the calculation in the Trial Balance
  of undistributed profits/losses account, and provide a footer note
  to the user explaining why will the ending balances will not be zero, but the
  period's total profit and loss.
- add an extra line for P&L application so that the trial balance zeroes in the
  balance, and total debits = total credits
- refactoring of the unaffected earnings reporting in the general ledger
  and trial balance.

JordiBForgeFlow committed:
remove account groups if hierarchy = 'none'

leemannd committed:
account_financial_report: set backgroud-color to table

psc committed:
Translated using Weblate (German)

Currently translated at 74.6% (217 of 291 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/de/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

JordiBForgeFlow committed:
[account_financial_report][IMP] Adds the following:
All reports:
- Rename field to hide accounts at 0 to 'hide_account_at_0'
Trial Balance:
- Add possibility to filter by hierarchy levels
- XLSX format will show the hierarchy levels in bold

General Ledger:
- Add the possibility to filter by analytic tags
- Fixes an error on the default date
Journal Ledger:
- The filter on Journals is now optional. If the user does not choose
a journal, by default it will display all journals.

Aged Partner Balance:
- Fixes an error on the default date

Commit:
[FIX] account_financial_report_qweb: Widget date for t-raw date (#483)

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 96.4% (291 of 302 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

JordiBForgeFlow committed:
[FIX+IMP] account_financial_report: new README structure + fixes in usability

dhongu committed:
Translated using Weblate (Romanian)

Currently translated at 39.7% (120 of 302 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/ro/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 21.5% (67 of 312 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/it/

JordiBForgeFlow committed:
Translated using Weblate (Spanish)

Currently translated at 99.7% (311 of 312 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

JordiBForgeFlow committed:
add proper titles

pedrobaeza committed:
[FIX] account_financial_report: 2 things:

* Fix account group level computation

  Depends was not correct for recomputing when needed + better algorithm

* Make hide details on 0 work properly

  * Passing values to general ledger was stripping some correct records
  * Computed field for hiding lines doesn't have proper dependencies nor is not
    taking into account float currency accuracy

JordiBForgeFlow committed:
[FIX] account_financial_report: multicompany+layout behaviour (#498)

* Fix wizards for proper multicompany behaviour.
* Fix layout issue

JordiBForgeFlow committed:
[account_financial_report]
- minor usability fixes for multicompany
- improve performance of general ledger using a new index

JordiBForgeFlow committed:
[account_financial_report] fix title formatting for all reports

wpichler committed:
[MIG] account_financial_report: Migration to 12.0

[IMP] Reformated one query to avoid one sql injection warning message - but no sql injection was possible here

Commit:
[IMP] account_financial_reports: Several improvements:

* Move to less files to compile variables
* hide contacts on partner
* Add reports to res partner actions
* initialize wizard from context
* initialize wizard by current fiscal year

MiquelRForgeFlow committed:
[FIX] account_financial_report: singleton error in aged partner balance report

Solves https://github.com/OCA/account-financial-reporting/issues/513

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (328 of 328 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

pedrobaeza committed:
[FIX] account_financial_report: Incorrect SQL query for cost centers

oca-travis committed:
[UPD] Update account_financial_report.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/

tv-openbig committed:
Translated using Weblate (German)

Currently translated at 85.7% (276 of 322 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/de/

cubells committed:
[IMP] account_financial_report: autoajust columns

qgroulard committed:
[FIX] Few fixes to solve warnings:

- _description in account_financial_report_abstract
- _description in vat.report.wizard
- Two fields using same label 'Accounts' in account.group
- Two fields using same label 'Account' in report_journal_ledger_move_line
- Two fields using same label 'Partner' in report_journal_ledger_move_line
- Two fields using same label 'Accounts' in report_trial_balance_account
- Two fields using same label 'Tax' in report_vat_report_taxtag
- Two fields using same label 'Tax' in report_vat_report_tax
- Two fields using same label 'Filter accounts' in general.ledger.report.wizard

JordiBForgeFlow committed:
account_financial_report: add option to hide parent hierarchy levels

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (323 of 323 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

Commit:
fix check value param self.hierarchy for account grouping

Commit:
Translated using Weblate (Italian)

Currently translated at 22.3% (72 of 323 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/it/

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (323 of 323 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_financial_report/es/

AdriaGForgeFlow committed:
[11.0][FIX] open Account Move Line view from Aged Partner Balance report

eLBati commented:
@feketemihai @mpanarin  see https://github.com/OCA/account-financial-reporting/pull/559#issuecomment-497226416 Could you tell me which email should I set ?

pedrobaeza commented:
Have you included all the mentioned PRs?

eLBati commented:
@qgroulard  I was not able to apply 8796d6f1077ed51161e205178b27475d7d186d99 
How did you generate that? Could you propose another PR to my branch please?

eLBati commented:
@pedrobaeza Merged PRs are incuded.
I put the TODO list in description. Please have a look

pedrobaeza commented:
OK, thanks for gathering this. Let's push for have it in v12!

mpanarin commented:
> @feketemihai @mpanarin see [#559 (comment)](https://github.com/OCA/account-financial-reporting/pull/559#issuecomment-497226416) Could you tell me which email should I set ?

@eLBati `mykhailopanarin@gmail.com`

eLBati commented:
@mpanarin thanks, fixed

eLBati committed:
[MIG] account_financial_report: Finalize migration to 12.0

* Move back to CSS and FIX

  NameError: variable @odoo-view-background-color is undefined in - on line 99, column 23:
  98 .o_account_financial_reports_page {
  99     background-color: @odoo-view-background-color;
  100     color: @odoo-main-text-color;

* Porting: parent_left does not exist anymore. Using parent_path + account code to sort report lines

* FIX js error with date object not converted to string while clicking on initial balance amount
  See https://user-images.githubusercontent.com/1033131/58337566-5d525c80-7e46-11e9-913a-3c3e0115fb3e.gif

* IMP style and metadata

* FIX tests about new date format and partner_id computed field of account.move

* FIX errors like

  2019-06-03 16:11:51,406 17857 ERROR dev_12_account_financial_report odoo.sql_db: bad query: b"\nDELETE FROM 'report_vat_report_tax'\nWHERE COALESCE(\n    write_date, create_date, (now() at time zone 'UTC'))::timestamp\n    < ((now() at time zone 'UTC') - interval '3600.0 seconds')\n"
  ERROR: syntax error at or near "'report_vat_report_tax'"
  LINE 2: DELETE FROM 'report_vat_report_tax'
                    ^
* IMP translation template and IMP Italian translation

* Remove useless comments
* Use AsIs to avoid SQL injection
* Use fields.Date methods
* Remove useless data from tests
* Improve comments
* Fix wizard.multi.charts.accounts
* Move _get_partner_ids_domain to abstract wizard
* Refactor default partners in wizard to use recordsets
* Improve js style

MiquelRForgeFlow commented:
@eLBati Please, remove https://github.com/OCA/account-financial-reporting/pull/554 from your TODO.

eLBati commented:
@mreficent done, thanks.

I have a problem with `account_financial_report/static/src/scss/report.scss`: I don't understand why it is not applied to PDF reports. Can anyone have a look?

About translations, I updated `account_financial_report.pot` and `it.po`

tate11 commented:
my opinion for PDF reports necessary inherit_id="web.report_assets_common"

eLBati commented:
@tate11 thanks! Fixed.

This is now ready for me

MiquelRForgeFlow reviewed (approved):
LGTM! :+1: 

Functionally tested

pedrobaeza commented on account_financial_report/view/report_template.xml:
```diff
@@ -0,0 +1,69 @@
+
+    <template id="account_financial_report_assets_common" name="account_financial_report_assets_common" inherit_id="web.assets_common">
+        <xpath expr="." position="inside">
+            <link rel="stylesheet" type="text/scss" href="/account_financial_report/static/src/scss/report.scss" />
```
Why do you need to change this if you have already changed https://github.com/OCA/account-financial-reporting/pull/559/files#diff-1d943c35c14f57134d9ad64496c5f9b5R6?

eLBati replied:
@pedrobaeza I just removed the SCSS from `account_financial_report.internal_layout` because useless (https://github.com/OCA/account-financial-reporting/compare/7130857bce83b818ac91fe01209f473e271a3759..7c8c6ad5406fa9318e5a7be70b346a0c99a96494)

Removing it from `web.assets_common` would make the HTML report style fail. Removing from `web.report_assets_common` would make the PDF  style fail

pedrobaeza replied:
But why don't you simply keep it as in current v11? I don't want to have a background in the screen for "View" option, and that's what is happening with scss compiled file. This is a large discussion and finally I have removed variables in v11, so no need to put scss, but simply css.

Any way, my fear putting this is that this alters other styles in the backend.

eLBati replied:
@pedrobaeza so you propose to move back to CSS removing the use of `$o-view-background-color` and `$o-main-text-color`, right?

pedrobaeza replied:
Yes, and keep the 2 assets links as they were.

eLBati replied:
Thanks, done

eLBati commented:
@OCA/accounting-maintainers what do you think?

pedrobaeza commented:
Reviewing this today for merging. Coverage is still decreasing a lot for such a critical module, but it's not your fault. Also I would want it to remove all the SQL mess, but one step at a time...

pedrobaeza commented on account_financial_report/tests/abstract_test.py:
```diff
@@ -0,0 +1,399 @@
+        return self
+
+    def _chart_template_create(self):
+        transfer_account_id = self.env['account.account.template'].create({
```
Don't create this account template, as it's now auto-created from prefix.

eLBati replied:
If I remove it, I got
```
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: ` Check if report HTML is correctly generated 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: Traceback (most recent call last): 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/account-financial-reporting/account_financial_report/tests/abstract_test.py", line 185, in setUp 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     self._add_chart_of_accounts() 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/account-financial-reporting/account_financial_report/tests/abstract_test.py", line 96, in _add_chart_of_accounts 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     self.chart.try_loading_for_current_company() 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "<decorator-gen-184>", line 2, in try_loading_for_current_company 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/odoo/odoo/api.py", line 372, in loop 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     result = [method(rec, *args, **kwargs) for rec in self] 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/odoo/odoo/api.py", line 372, in <listcomp> 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     result = [method(rec, *args, **kwargs) for rec in self] 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/odoo/addons/account/models/chart_template.py", line 178, in try_loading_for_current_company 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     self.load_for_current_company(15.0, 15.0) 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/odoo/addons/account/models/chart_template.py", line 252, in load_for_current_company 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     company.transfer_account_id = self.env['account.account'].search([('code', '=like', self.transfer_account_code_prefix + '%')])[0] 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `   File "/home/elbati/workspace/odoo/instances/odoo12/parts/odoo/odoo/models.py", line 5121, in __getitem__ 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: `     return self._browse((self._ids[key],), self.env) 
2019-06-13 14:19:06,488 2518 ERROR demo_12 odoo.addons.account_financial_report.tests.test_aged_partner_balance: ` IndexError: tuple index out of range
```

pedrobaeza replied:
It should be created in https://github.com/odoo/odoo/blob/a526ff3f70c472fcd7647e1b7b75dcfa9d539209/addons/account/models/chart_template.py#L130

I'll investigate locally and change it at the same time as I squash some commits before merging.

pedrobaeza replied:
OK, I'm not able to find the reason, but on l10n_es works without this. Let's move on as no real problem to let that as they are tests.

pedrobaeza commented on account_financial_report/tests/abstract_test.py:
```diff
@@ -0,0 +1,399 @@
+            'currency_id': self.ref('base.USD'),
+            'transfer_account_code_prefix': '000',
+        })
+        transfer_account_id.update({
```
No need of this (see above).

pedrobaeza commented on account_financial_report/report/abstract_report_xlsx.py:
```diff
@@ -188,6 +190,8 @@ def write_line(self, line_object):
         for col_pos, column in self.columns.items():
             value = getattr(line_object, column['field'])
+            if isinstance(value, datetime.date):
+                value = datetime.datetime.strftime(value, DEFAULT_SERVER_DATE_FORMAT)
```
Better to use `fields.Date.to_string`

eLBati replied:
Done

pedrobaeza commented on account_financial_report/report/abstract_report.py:
```diff
@@ -13,9 +13,9 @@ def _transient_clean_rows_older_than(self, seconds):
     < ((now() at time zone 'UTC') - interval %s)
 """
-        self.env.cr.execute(query, ("%s seconds" % seconds,))
+        self.env.cr.execute(query, (self._table, "%s seconds" % seconds,))
```
Use `AsIs(self._table)` for avoiding problems and proper identification. You need at the beginning:
```python
from psycopg2.extensions import AsIs
```

eLBati replied:
Done

pedrobaeza commented on account_financial_report/menuitems.xml:
```diff
@@ -58,6 +58,7 @@
         groups="base.group_erp_manager"
         />
 
+    <!--
```
Remove this instead of commenting

eLBati replied:
Done

pedrobaeza commented on account_financial_report/menuitems.xml:
```diff
@@ -58,6 +58,7 @@
         groups="base.group_erp_manager"
```
What does this mean of hiding PDF reports? Isn't in v12 the reports already gone?

eLBati replied:
They are. Fixing

pedrobaeza commented on account_financial_report/tests/test_general_ledger.py:
```diff
@@ -87,11 +88,11 @@ class TestGeneralLedgerReport(common.TransactionCase):
-        self.previous_fy_date_end = '2015-12-31'
-        self.fy_date_start = '2016-01-01'
-        self.fy_date_end = '2016-12-31'
+        self.before_previous_fy_year = datetime.strptime('2014-05-05', '%Y-%m-%d')
```
Use `fields.Date.from_string` in all of this.

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/general_ledger_wizard.py:
```diff
@@ -121,7 +122,7 @@ def _compute_fy_start_date(self):
             date = fields.Datetime.from_string(wiz.date_from)
             res = self.company_id.compute_fiscalyear_dates(date)
-            wiz.fy_start_date = res['date_from']
+            wiz.fy_start_date = datetime.strftime(res['date_from'], DEFAULT_SERVER_DATE_FORMAT)
```
Use `fields.Date.to_string`

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/trial_balance_wizard.py:
```diff
@@ -98,7 +98,7 @@ def _compute_fy_start_date(self):
             date = fields.Datetime.from_string(wiz.date_from)
             res = self.company_id.compute_fiscalyear_dates(date)
-            wiz.fy_start_date = res['date_from']
+            wiz.fy_start_date = datetime.strftime(res['date_from'], DEFAULT_SERVER_DATE_FORMAT)
```
Use `fields.Date.to_string`

eLBati replied:
Done

pedrobaeza commented on account_financial_report/report/abstract_report.py:
```diff
@@ -6,16 +6,17 @@
-    < ((now() at time zone 'UTC') - interval %s)
-"""
+        query = (
+            "DELETE FROM " + self._table +
```
Avoid here SQL injection as done in other change

eLBati replied:
How is this affected by SQL injection? `self._table` doesn't come from user input

pedrobaeza replied:
Someone can manipulate that python variable. In other place you have replace it passing that parameter in the list. I suggested to put it with `AsIs` method for avoiding the need to disable the linter.

eLBati replied:
Like this
```
        query = AsIs(
            "DELETE FROM " + self._table +
            " WHERE COALESCE("
            "write_date, create_date, (now() at time zone 'UTC'))"
            "::timestamp < ((now() at time zone 'UTC') - interval %s)"
        )
```
?

> Someone can manipulate that python variable

I wonder how

> for avoiding the need to disable the linter

It is not disabled here. Am I missing something?

pedrobaeza replied:
No, I mean:
```python
query = "DELETE FROM %s..."
cr.execute(query, (AsIs(self._table), ...))
```

eLBati replied:
Done

pedrobaeza commented on account_financial_report/report/abstract_report_xlsx.py:
```diff
@@ -188,6 +190,9 @@ def write_line(self, line_object):
         for col_pos, column in self.columns.items():
             value = getattr(line_object, column['field'])
+            if isinstance(value, datetime.date):
+                value = datetime.datetime.strftime(
```
`fields.Date.to_string`

pedrobaeza commented on account_financial_report/report/open_items.py:
```diff
@@ -638,6 +639,7 @@ def _compute_partners_and_accounts_cumul(self):
         self._compute_account_cumul()
 
     def _compute_partner_cumul(self):
+        # pylint: disable=sql-injection
```
Instead of disabling pylint, use `AsIs` for passing a full expression without warnings

eLBati replied:
To avoid disabling pylint, I think here we should remove variables like `where_condition_partner_by_account` and write the string directly at the end of `query_computer_partner_residual_cumul` and `query_compute_partners_due_cumul`, right?

pedrobaeza replied:
Well, I haven't seen full code right now, but you know the basis. Do what you consider best :smiley:

eLBati replied:
Done

pedrobaeza commented on account_financial_report/report/trial_balance.py:
```diff
@@ -75,7 +75,7 @@ class TrialBalanceReportAccount(models.TransientModel):
     hide_line = fields.Boolean(compute='_compute_hide_line')
     # Data fields, used to keep link with real object
-    sequence = fields.Integer(index=True, default=1)
+    sequence = fields.Char(index=True, default='1')
```
Put a comment in code about why is this a char (due to `parent_path`)

pedrobaeza commented on account_financial_report/tests/abstract_test.py:
```diff
@@ -109,15 +108,7 @@ def _add_chart_of_accounts(self):
         })
         self.with_context(
             company_id=self.company.id, force_company=self.company.id)
-        wizard = self.env['wizard.multi.charts.accounts'].create({
```
Not sure if a previous error in this method is still there in v12. See https://github.com/OCA/bank-payment/commit/9c5a98ffbe72531098b7c10c676b2c616284fd9b for details.

You can test the problem when having other chart template installed in the database doing the tests.

Any way, previous code is missing explicitly in the creation dictionary `'sale_tax_id' : False` and `'purchase_tax_id': False`, which avoids the problem.

eLBati replied:
I don't understand, tests are green. How could another chart be present?

pedrobaeza replied:
When you are executing your tests in a customer environment CI (aka more or less integration tests) that has for example `l10n_es` installed.

eLBati replied:
I installed `l10n_es` in a demo DB, executed tests of `account_financial_report` and no error was raised

pedrobaeza commented on account_financial_report/tests/test_general_ledger.py:
```diff
@@ -87,11 +88,13 @@ class TestGeneralLedgerReport(common.TransactionCase):
-        self.previous_fy_date_end = '2015-12-31'
-        self.fy_date_start = '2016-01-01'
-        self.fy_date_end = '2016-12-31'
+        self.before_previous_fy_year = datetime.strptime(
```
`fields.Date.from_string('2014-05-05')`
or even better:
`date(2014, 5, 5)`

pedrobaeza commented on account_financial_report/wizard/aged_partner_balance_wizard.py:
```diff
@@ -60,8 +60,10 @@ def onchange_company_id(self):
                 '|', ('company_id', '=', self.company_id.id),
-                ('company_id', '=', False)]
+                ('company_id', '=', False),
+                ('parent_id', '=', False)]
```
This part of the domain is not correct at all: you can have subsidiaries marked as company (`is_company`), instead of individual, having a parent_id.  See the computation of `commercial_parent_id` as reference:

https://github.com/odoo/odoo/blob/3c45f08825b1a0de35f2027f7cb53977f2c11f08/odoo/addons/base/models/res_partner.py#L276

A possible domain would be:
```
'|',
('parent_id', '=', False),
('is_company', '=', True),
```

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/general_ledger_wizard.py:
```diff
@@ -80,15 +85,45 @@ class GeneralLedgerReportWizard(models.TransientModel):
+                filtered(lambda p: p.parent_id)
+
+            partner_ids = set(partner_ids) - set(corp_partners.ids)
+            partner_ids |= set(corp_partners.mapped('parent_id.id'))
```
`partner_ids |= set(corp_partners.mapped('commercial_partner_id.id'))`

But it's better to work full method with recordsets:
```python
partners = self.env['res.partner'].browse(context['active_ids'])
corp_partners = partners.filtered('parent_id')
return partners - corp_partners + corp_partners.mapped('commercial_partner_id')
```

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/general_ledger_wizard.py:
```diff
@@ -80,15 +85,45 @@ class GeneralLedgerReportWizard(models.TransientModel):
             date = fields.Datetime.from_string(wiz.date_from)
             res = self.company_id.compute_fiscalyear_dates(date)
-            wiz.fy_start_date = res['date_from']
+            wiz.fy_start_date = datetime.strftime(
```
`fields.Date.to_string`

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/general_ledger_wizard.py:
```diff
@@ -135,8 +170,10 @@ def onchange_company_id(self):
                 '|', ('company_id', '=', self.company_id.id),
-                ('company_id', '=', False)]
+                ('company_id', '=', False),
+                ('parent_id', '=', False)]
```
The same about the domain

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/open_items_wizard.py:
```diff
@@ -44,14 +44,32 @@ class OpenItemsReportWizard(models.TransientModel):
+    def _default_partners(self):
+        context = self.env.context
+
+        if context.get('active_ids') and context.get('active_model') \
```
The same about how to get partners through recordsets

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/open_items_wizard.py:
```diff
@@ -73,8 +91,10 @@ def onchange_company_id(self):
                 '|', ('company_id', '=', self.company_id.id),
-                ('company_id', '=', False)]
+                ('company_id', '=', False),
+                ('parent_id', '=', False)]
```
The same about domain

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/trial_balance_wizard.py:
```diff
@@ -98,7 +99,8 @@ def _compute_fy_start_date(self):
             date = fields.Datetime.from_string(wiz.date_from)
             res = self.company_id.compute_fiscalyear_dates(date)
-            wiz.fy_start_date = res['date_from']
+            wiz.fy_start_date = datetime.strftime(
```
`fields.Date.to_string`

eLBati replied:
Done

pedrobaeza commented on account_financial_report/wizard/trial_balance_wizard.py:
```diff
@@ -138,8 +140,10 @@ def onchange_company_id(self):
                 '|', ('company_id', '=', self.company_id.id),
-                ('company_id', '=', False)]
+                ('company_id', '=', False),
+                ('parent_id', '=', False)]
```
The same about the domain. As this domain is repeated through a lot of places, I think it should be extracted to a method in the abstract model or similar.

eLBati replied:
Done

eLBati commented:
@pedrobaeza please have a look at fdaa4848d44bc40523c7d7a4e970ead495b6107b
Thanks!

pedrobaeza commented:
Thanks all for your work!
